src/sentry/dynamic_sampling/utils.py

```diff
@@ -43,3 +54,54 @@ def generate_uniform_rule(project: Project) -> UniformRule:
         },
         "id": UNIFORM_RULE_RESERVED_ID,
     }
+
+
+class DynamicSamplingFeatureMultiplexer:
+    """
+    This class is used to route DS behaviour according to the feature flags listed here
+    Essentially the logic is as follows:
+    - The `organizations:server-side-sampling` feature flag is the main flag enabled for dynamic sampling both in
+    sentry and in relay
+    - The  `organizations:server-side-sampling-ui` feature flag is the flag that enables the old dynamic sampling
+    behaviour which needs to be supported for backwards compatibility but is deprecated
+    - The `organizations:dynamic-sampling-basic` feature flag is the flag that enables the new adaptive sampling
+    """
+
+    def __init__(self, project, request):
```

Comment:
```suggestion
    def __init__(self, project, user):
```
looks like we need only `user` here.

Reply:
Yes agreed will fix